### PR TITLE
[WIP] First pass at adding a shell.

### PR DESCRIPTION
**Pull request by duckinator**

### CODE REVIEW

**Comment by danopia** [comment]
This is cool :) does the prompt get printed for follow up commands? I only see one print call.

**Comment by duckinator** [comment]
in theory, yes. in practice, it kernel panics when i try to run any commands lmfao

File: src/libraries/shell/src/main.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        for (size_t i = 0; i < result->num_pieces; i++) {
+            print(result->pieces[i]);
+            if (i < (result->num_pieces - 1)) {
+                print(" ");
+            } else {
+                print("\n");
+            }
+        }
+    } else {
+        print("Error: Unkown command: ");
```
[the remark this answers is not in the document]
Comment (duckinator) [reply src/libraries/shell/src/main.c]: thanks for pointing that out!

**Pull request merged**